### ISSUE TITLE
[M7] Tiny OCM protected stress, scaling, and matched lifetime-cost comparison

### ISSUE THREAD

**Issue by SzeChunYiu** [SzeChunYiu/ORION-OCM#9]
# M7 — Protected language stress test and matched-parent comparison

**Parent:** #1  
**Depends on:** #8 (`LANGUAGE_KSO_ALPHA`)  
**Gate:** closes the language-first programme and unlocks M8+  
**Admissible terminals:** `OCM_LANGUAGE_RESIDUAL_SUPPORTED`, `PARENT_SUFFICIENT`, `CANNOT_CHECK`

## Goal

Subject Conversational OCM Alpha to a frozen, hostile, multi-axis evaluation that determines what—if anything—the Machine Epistemics architecture contributes beyond the strongest faithful parents under matched information and resource conditions.

M7 is the scientific gate. M6 proves that a machine exists; M7 determines what the evidence supports.

---

## 1. Pre-registration before protected outcome access

Freeze in a committed protocol:

```text
research questions
hypotheses
primary/secondary metrics
datasets + versions
train/dev/protected splits
comparators
information budgets
compute/memory/tool budgets
statistical tests
minimum effect sizes where used
hostile mutations
stopping rules
admissible terminals
```

Hash the protocol and dataset manifests before protected evaluation.

No metric, split, comparator, or exclusion rule may be changed after protected outcome access without opening a new versioned study.

---

## 2. Research questions

At minimum test:

### RQ1 — Conversational competence

Can OCM sustain correct/coherent multi-turn English conversation within its declared knowledge scope?

### RQ2 — Cognitive grounding

Does the response derive from task/KSO state rather than a language component that can independently reconstruct protected answers?

### RQ3 — Continual acquisition

Can OCM acquire new language knowledge/procedures after deployment more locally/sample-efficiently than matched parents?

### RQ4 — Correction/revocation

Does OCM correct affected behavior without unnecessary global retraining or unrelated regression?

### RQ5 — Ambiguity/unknown calibration

Does explicit epistemic state improve clarification and honest-unknown behavior?

### RQ6 — Negative transfer

Does OCM correctly refuse/specialize misleading transfer rather than blindly reuse similar constructions?

### RQ7 — Whole-system resources

What compute, memory, storage, indexing, interaction and update costs buy the observed behavior?

---

### 3.1 Grammatical competence — BLiMP

Use the official BLiMP release/version pinned at execution. Evaluate the relevant 67 paradigms / 67k minimal pairs or the full official evaluation pipeline if licensing/version remains compatible.

Report per-phenomenon accuracy, not only aggregate:

- agreement;
- argument structure;
- binding;
- control/raising;
- determiners/quantifiers;
- ellipsis;
- filler-gap;
- irregular forms;
- island effects;
- NPI licensing;
- syntax/semantics interface phenomena.

If OCM does not expose sentence likelihoods, define and freeze an admissibility/interpretability decision protocol before testing rather than retrofitting a favorable metric.

### 3.2 Sample-efficient language learning — BabyLM 2026

Pin official 2026 datasets/evaluation code.

Required runs where feasible:

- ≤10M-word Strict-Small-aligned regime;
- ≤100M-word Strict-aligned regime;
- optional multilingual preflight under current rules.

All non-text supervision (aligned meanings, explicit lessons, interaction, images, teacher feedback) must be quantified as an additional information channel.

### 3.3 Real syntax/domain shift — UD English EWT

Pin current release at execution (current official page reports UD 2.18 and CC BY-SA 4.0). Evaluate parsing/interpretation robustness across blog, social, review, email and web genres.

### 3.4 Developmental/interaction language — CHILDES

Use only selected corpora whose current TalkBank access terms are satisfied. Record access level and do not commit restricted transcripts to the repo.

Evaluate incremental acquisition and repair/reference patterns, not biological plausibility claims.

### 3.5 Protected OCM conversation suite

Freeze human-authored multi-turn conversations covering:

- follow-ups and delayed references;
- corrections;
- ambiguity;
- contradiction;
- unknown information;
- explanation/compare/summarize;
- style shifts;
- new word/construction teaching;
- post-restart continuation;
- revocation mid-dialogue;
- adversarially misleading surface similarity.

### 3.6 Bounded-knowledge factual suite

Generate protected questions from the exact frozen bounded-world KSO plus adversarial out-of-scope questions.

Measure factual accuracy and honest-unknown behavior separately.

---

## 4. Strongest faithful parent construction

Do not compare OCM only with weak templates.

Build the strongest feasible comparator product with the same task-relevant powers. Candidate components may include:

```text
Transformer/recurrent language learner trained on matched corpus
retrieval/vector memory over the same knowledge set
dialogue-state memory
post-deployment fine-tuning / adapter update
structured parser or semantic parser
explicit verification/retrieval where OCM also has it
same bounded knowledge corpus
same interaction/test-time budget
```

The comparator may combine known parents; the purpose is to subtract existing capability, not protect the OCM narrative.

### External frontier reference

Also evaluate a current frontier chat model at execution time for product-quality context. This arm is **not matched** if it has enormous prior training exposure and must never be used as the causal comparator for OCM sample-efficiency claims.

---

## 5. Information accounting

For every arm record:

```text
training words/tokens
gold syntactic annotations
meaning graphs / aligned semantics
explicit grammar lessons
dictionary entries
teacher feedback
interaction turns
grounded observations
knowledge-base facts
retrieval documents
protected-test exposure = 0
```

Create an `INFORMATION_BUDGET_RECEIPT` for every run.

A comparison is labeled `MATCHED` only if the tested difference is the declared mechanism rather than extra information.

---

## 6. Resource accounting

Track separately:

```text
training/update FLOPs or best available compute proxy
CPU/GPU wall time
peak memory
persistent storage
index size
KSO object/relation count
query latency
external IO/tool calls
post-deployment update cost
retraining cost
human/teacher interaction count
```

Do not hide KSO storage/retrieval cost while reporting a small trainable core.

---

## 7. Primary outcome vector

Report without collapsing to a single intelligence score:

```text
semantic/task correctness
grammatical competence
multi-turn task success
reference/correction accuracy
uncertainty calibration
honest-unknown accuracy
post-deployment acquisition rate
sample efficiency
retention/forgetting
negative-transfer harm
transfer precision
revocation/correction locality
human-rated naturalness/helpfulness
latency/memory/storage/update cost
```

### Required derived metrics

Where appropriate include:

```text
LifetimeLearningEfficiency
NetTransfer = positive transfer gain - harmful transfer - declared resource penalty
CorrectionLocality = affected behavior corrected / unrelated behavior changed
```

Keep raw components visible.

---

## 8. Post-deployment learning challenge

Freeze multiple unknown constructions/lexemes after base training.

For each:

1. baseline test before teaching;
2. provide fixed lesson/demo/interaction budget;
3. test immediate acquisition;
4. test unseen compositional instances;
5. run retention suite;
6. restart process;
7. retest;
8. revoke teaching evidence;
9. verify dependent behavior disappears/reopens;
10. relearn from alternate support.

Compare against parent methods such as fine-tuning, in-context memory, retrieval memory and adapter updates under matched lesson exposure.

---

## 9. Negative-transfer challenge

Protected pairs must include:

- structurally similar but semantically incompatible construction;
- English vs controlled SOV word-order transfer;
- polysemous lexical senses;
- idiom vs literal phrase;
- style/register vs grammatical requirement;
- high-frequency incorrect corpus pattern;
- local conversational convention vs global language rule.

Measure:

```text
transfer attempted
transfer beneficial/harmful
confidence/applicability score
recovery after counterexample
```

A system that never transfers is not good; a system that transfers everything is not good.

---

## 10. Answer-laundering / hidden-solver audit

Independently inspect every language/evaluation component for pathways that could reconstruct protected answers without KSO cognition.

Required adversarial audits:

- replace cognitive conclusion with a random/incorrect meaning while keeping user text; renderer must follow supplied meaning, not infer answer from prompt;
- remove relevant KSO knowledge; language component must not answer from its own hidden corpus/model if the mechanism contract prohibits that;
- swap admissible codecs/parsers; underlying protected conclusion should remain equivalent where translator-invariance is claimed;
- inject a gold-answer field into a forbidden preprocessing channel; checker must detect it.

Report all external pretrained components and exactly what information they may contain.

---

## 11. Human evaluation protocol

Blind/randomized raters separately score:

- intent understanding;
- correctness;
- relevance/helpfulness;
- coherence with history;
- grammaticality;
- naturalness;
- uncertainty appropriateness;
- correction quality.

Use enough raters/items for predefined confidence intervals or statistical power. Freeze exclusion/quality-control rules before viewing system identity outcomes.

Do not tell raters which output is OCM.

---

## 12. Statistical analysis

Predefine tests per metric:

- paired bootstrap/permutation for paired task accuracy where appropriate;
- exact/binomial or McNemar-type tests for paired binary outcomes;
- confidence intervals for rates;
- inter-rater agreement and mixed-effects/appropriate models for human ratings if warranted;
- multiple-comparison handling for large phenomenon families.

Report effect sizes and confidence intervals, not only p-values.

A non-significant advantage is not superiority.

---

## 13. Required ablations

At minimum:

```text
OCM full
- warrant/revocation
- explicit dialogue workspace
- construction abstraction
- active clarification
- raw-corpus induction
- grounded/aligned learning
- thought↔language reopen loop
- KSO navigation (replace with retrieval parent)
- local continual update (replace with retraining)
```

Ablations must preserve as much other machinery/information as possible.

---

## 14. Hostile mutations

Required:

- training/test contamination;
- protected split selected after tuning;
- comparator denied annotations OCM receives;
- resource meter excludes retrieval index;
- revoked construction remains in compiled cache;
- LLM/teacher gives hidden semantic gold during mechanism evaluation;
- uncertainty surface language hard-coded independent of belief state;
- human evaluator sees system identity;
- failed examples removed post hoc;
- parent intentionally under-tuned;
- stopping rule ends when OCM leads.

Every mutant should be detectable by automation or independent audit where feasible.

---

### `OCM_LANGUAGE_RESIDUAL_SUPPORTED`

Allowed only if a pre-registered capability/resource residual survives the strongest faithful matched parent and relevant ablations, with uncertainty/effect-size requirements met.

### `PARENT_SUFFICIENT`

Use if the strongest faithful parent matches/explains the claimed effect under matched conditions.

### `CANNOT_CHECK`

Use for data/license/tooling/compute or protocol failures that prevent a valid test.

Mixed outcomes are allowed: individual claims can be parent-sufficient while others retain a residual.

---

## 16. Deliverables

- [ ] frozen pre-registration/protocol
- [ ] dataset/version/license manifest
- [ ] information-budget receipts
- [ ] resource receipts
- [ ] BLiMP adapter/results
- [ ] BabyLM 10M/100M run(s) or explicit CANNOT_CHECK
- [ ] UD EWT evaluation
- [ ] CHILDES subset study if authorized
- [ ] protected conversation/knowledge suites
- [ ] strongest-parent implementation
- [ ] external frontier reference evaluation
- [ ] post-deployment learning challenge
- [ ] negative-transfer challenge
- [ ] ablation suite
- [ ] answer-laundering audit
- [ ] blinded human evaluation
- [ ] statistical report
- [ ] final claim-by-claim terminal table

---

## Exit gate / unlock rule

M8+ implementation remains locked until this issue closes with an admissible scientific terminal and the result is linked back to #1.

Closing M7 does **not** require OCM to win. An honest `PARENT_SUFFICIENT` result still completes the language programme scientifically and tells us which machinery to simplify before expanding.

Terminal: one of

```text
OCM_LANGUAGE_RESIDUAL_SUPPORTED
PARENT_SUFFICIENT
CANNOT_CHECK
```

**Comment by SzeChunYiu** [comment]
## Primary M7 scientific claim refinement — matched lifecycle cost

M7's primary economic/scaling claim is now bounded to **cost at equivalent bounded competence**, not comparison with total frontier-LLM pretraining cost.

For every matched arm measure:

```text
C_initial
C_new_capability
C_correction
C_retention
C_inference
C_memory/index
C_retrieval/navigation
C_verification
C_human/teacher_interaction
```

and report:

```text
C_lifetime = C_initial
           + Σ C_update
           + C_inference
           + C_memory/index
           + C_retrieval/navigation
           + C_verification
           + C_interaction
```

Primary matched questions:

1. At the same bounded language/task competence, does Tiny OCM require less total information/resource cost than the strongest small faithful parent?
2. After prior related learning, is the cost of acquiring capability `n+1` lower than the parent's adaptation cost?
3. Does that advantage survive memory/index/navigation/verification overhead?
4. Does KSO query/update/revocation cost scale tractably over the tested size range?

Use small Transformers/recurrent models, structured grammar/construction learners, and retrieval/memory systems as principal matched parents. A frontier chat model remains a **non-matched quality reference only**.

A valid positive conclusion is limited to the measured task ecology/range, e.g. `favorable lifetime-cost/scaling residual on the tested tiny prototype`. Do not extrapolate to frontier-scale training cost without later evidence.

**Comment by SzeChunYiu** [comment]
M7 study is up as a PR (branch m7/protected-comparison). Pre-registered (hash-bound) study; V1 run = DEV_CALIBRATION (three defects fixed after outcome access, ledger S22–S24); V2 protected run against the strongest faithful matched parent built here: RQ1 conversations 53/54 vs 33/54 (n=54, TOST) → OCM_LANGUAGE_RESIDUAL_SUPPORTED with mechanism attribution CANNOT_CHECK; RQ1 factual / RQ3 / RQ5 / RQ6 CANNOT_CHECK (n below the pre-registered minimum 40); RQ2 laundering audit 0 incidents; BLiMP / UD EWT coverage 0 → CANNOT_CHECK; BabyLM / CHILDES / human rating CANNOT_CHECK. Report docs/M7_PROTECTED_COMPARISON_REPORT.md with the revival backlog (coverage via UD structural alignment, suites at n ≥ 40, neural comparator, human rating).

**Comment by SzeChunYiu** [comment]
M7 terminal (claim by claim, as the pre-registration permits) — PR #22 merged with a merge commit (de5f01f). Receipt docs/provenance/M7_RECEIPT_V1.json; report docs/M7_PROTECTED_COMPARISON_REPORT.md. RQ1 conversations: OCM_LANGUAGE_RESIDUAL_SUPPORTED vs the strongest faithful matched parent built here (53/54 vs 33/54, n=54, TOST at δ=0.05) with mechanism attribution CANNOT_CHECK; RQ1 factual, RQ3, RQ5, RQ6: CANNOT_CHECK (n below the pre-registered minimum 40); RQ2 laundering audit 0 incidents; PARENT_SUFFICIENT where ablations were equivalent; BLiMP / UD EWT: CANNOT_CHECK (coverage 0); BabyLM / CHILDES / human rating / frontier reference: CANNOT_CHECK. V1 = DEV_CALIBRATION; V2 = the protected study. Admissible mixed terminal: closing #9 and unlocking M8+.
